Automattic/Gravatar-SDK-Android
# Quick Editor - Alt text feature foundations

Closes #497 

### Description

This PR implements the foundations of the `Alt Text` feature. It doesn't include all of the functionality, as some parts still need to be implemented/improved. [See sub-issues](https://github.com/Automattic/Gravatar-SDK-Android/issues/404)

I'm opening this PR for two reasons:
- I think we can parallelize the missing work
- To keep this PR as small as possible

The basic functionality, which is changing the alternative text for an image, is implemented. 
You should be able to see the confirmation toast and the error toast when something fails.

<img width="410" alt="image" src="https://github.com/user-attachments/assets/806bf6aa-783e-45ed-8f64-0eb4df3ad4fc" />

### Testing Steps

1. Open the QE and the options for any avatar
2. Modify the alt text
3. Verify the `Save` is enabled when the value is different from the original alt text
4. Save a new alt text
5. Verify the QE navigates to the Avatar picker
6. Close the QE bottom sheet and reopen it
7. Open the alt text for the previous image and verify you see the new alt text
8. Try to change it again but force an error (network error is enough)
9. Verify you see the error toast
10. Open the alt text again and verify you see the previous alt text (not the one that produces the error)
11. Verify dark/light mode
12. Verify orientation changes
